proposals/0023-asynchronous-program-execution-and-broadcast.md

```diff
@@ -0,0 +1,148 @@
+---
+simd: '0023'
+title: Asynchronous Program Execution and Broadcast
+authors:
+  - Anatoly Yakovenko
+category: Standard/Meta
+type: Core
+status: Draft
+created: 2024-04-29
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+This feature changes how the ledger is broadcast and executed. It
+separates proposing blocks full of user transactions from blocks
+with votes.  It allows for N concurrent builders of user transaction
+blocks, and it allows for asynchronous execution of user blocks.
+
+## Motivation
+
+1. Single leader for user transactions is a bottleneck.  Leaders
+are in a single spot in the world, while clients are all over the
+world. Latency is therefore based on how close a client is to a
+leader.  Leaders also have to spend a ton of resources on prioritization
+of transactions.
+
+2. Executing programs before voting is a bottleneck. Fork choice
+doesn't depend on program execution, and delaying voting creates
+forks and also delays the next block producer from starting.
+
+## Alternatives Considered
+
+APEX design that was proposed in https://github.com/solana-labs/solana/pull/24127
+
+## New Terminology
+
+* UserBlock - a block full of none vote transactions.
+
+* UserBlockEntry - a hash of the block that was proposed
+
+* UserBlockSlot - a slot measuring half a normal slot in ticks.
+
+* Leader - the current leader for the slot that will propose a PoH
+ledger full of Votes and UserBlockEntry
+
+* Builder - a node that is scheduled to propose a block with non
```

Comment:
proposer?

Reply:
Isn't builder a better name? Proposer is too generic, since the leader also proposes a block

Reply:
a proposer proposes the block to the network that others vote on. in this case i think they'd be proposing the ordering?